js/Views/IView.ts

```diff
@@ -4,6 +4,4 @@ import {MigrationData} from "../Data/MigrationPatterns";
 export interface IView {
 
     readonly curYear: number;
-    readonly currentData: MigrationData;
```

Comment:
I see you've dropped this type and you're using a StateIndicators type.  Maybe the interface here is overkill for the project and we can just create some global state context that each view reads and updates on when a year is changed.